### PR TITLE
docs: Clarify locale properties on `context` object in data-fetching

### PR DESCRIPTION
**Pull request by TyMick**
Until I started looking into the source code to verify, I thought that `locale`, `locales`, and `defaultLocale` were being parsed from the [`Accept-Language`][1] header somehow. This'll help other people like me who are reading the [Data Fetching](https://nextjs.org/docs/basic-features/data-fetching) page but weren't already aware of Next.js's [Internationalized Routing](https://nextjs.org/docs/advanced-features/i18n-routing) features.

[1]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language

## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

File: docs/basic-features/data-fetching.md [hunk 1]
```diff
@@ -66,9 +66,9 @@ The `context` parameter is an object containing the following keys:
 - `params` contains the route parameters for pages using dynamic routes. For example, if the page name is `[id].js` , then `params` will look like `{ id: ... }`. To learn more, take a look at the [Dynamic Routing documentation](/docs/routing/dynamic-routes.md). You should use this together with `getStaticPaths`, which we’ll explain later.
 - `preview` is `true` if the page is in the preview mode and `undefined` otherwise. See the [Preview Mode documentation](/docs/advanced-features/preview-mode.md).
 - `previewData` contains the preview data set by `setPreviewData`. See the [Preview Mode documentation](/docs/advanced-features/preview-mode.md).
-- `locale` contains the active locale (if enabled).
-- `locales` contains all supported locales (if enabled).
-- `defaultLocale` contains the configured default locale (if enabled).
+- `locale` contains the active locale (if you've enabled [Internationalized Routing](/docs/advanced-features/i18n-routing.md)).
+- `locales` contains all supported locales (if you've enabled Internationalized Routing).
```
Comment (leerob) [inline docs/basic-features/data-fetching.md:70]: Might as well link all instances of Internationalized Routing here, yeah?

Comment (timneutkens) [reply docs/basic-features/data-fetching.md:70]:
```suggestion
- `locales` contains all supported locales (if you've enabled [Internationalized Routing](/docs/advanced-features/i18n-routing.md)).
```

File: docs/basic-features/data-fetching.md [hunk 2]
```diff
@@ -666,9 +666,9 @@ The `context` parameter is an object containing the following keys:
 - `preview`: `preview` is `true` if the page is in the preview mode and `false` otherwise. See the [Preview Mode documentation](/docs/advanced-features/preview-mode.md).
 - `previewData`: The preview data set by `setPreviewData`. See the [Preview Mode documentation](/docs/advanced-features/preview-mode.md).
 - `resolvedUrl`: A normalized version of the request URL that strips the `_next/data` prefix for client transitions and includes original query values.
-- `locale` contains the active locale (if enabled).
-- `locales` contains all supported locales (if enabled).
-- `defaultLocale` contains the configured default locale (if enabled).
+- `locale` contains the active locale (if you've enabled [Internationalized Routing](/docs/advanced-features/i18n-routing.md)).
+- `locales` contains all supported locales (if you've enabled Internationalized Routing).
```
Comment (timneutkens) [inline docs/basic-features/data-fetching.md:670]:
```suggestion
- `locales` contains all supported locales (if you've enabled [Internationalized Routing](/docs/advanced-features/i18n-routing.md)).
```

File: docs/basic-features/data-fetching.md [hunk 3]
```diff
@@ -666,9 +666,9 @@ The `context` parameter is an object containing the following keys:
...
+- `defaultLocale` contains the configured default locale (if you've enabled Internationalized Routing).
```
Comment (timneutkens) [inline docs/basic-features/data-fetching.md:671]:
```suggestion
- `defaultLocale` contains the configured default locale (if you've enabled [Internationalized Routing](/docs/advanced-features/i18n-routing.md)).
```

**Review by timneutkens — APPROVED** [review]

**Comment by TyMick** [comment]
🚀 Thanks for the revisions!

**Pull request merged**